capture-server/tests/events.rs

```diff
@@ -59,14 +62,98 @@ async fn it_captures_a_batch() -> Result<()> {
     assert_eq!(StatusCode::OK, res.status());
 
     assert_json_include!(
-        actual: topic.next_event()?,
+        actual: main_topic.next_event()?,
         expected: json!({
             "token": token,
             "distinct_id": distinct_id1
         })
     );
     assert_json_include!(
-        actual: topic.next_event()?,
+        actual: main_topic.next_event()?,
+        expected: json!({
+            "token": token,
+            "distinct_id": distinct_id2
+        })
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn it_captures_a_batch() -> Result<()> {
+    setup_tracing();
+    let token = random_string("token", 16);
+    let distinct_id1 = random_string("id", 16);
+    let distinct_id2 = random_string("id", 16);
+
+    let main_topic = EphemeralTopic::new().await;
+    let histo_topic = EphemeralTopic::new().await;
+    let server = ServerHandle::for_topics(&main_topic, &histo_topic).await;
+
+    let event = json!({
+        "token": token,
+        "batch": [{
+            "event": "event1",
+            "distinct_id": distinct_id1
+        },{
+            "event": "event2",
+            "distinct_id": distinct_id2
+        }]
+    });
+    let res = server.capture_events(event.to_string()).await;
+    assert_eq!(StatusCode::OK, res.status());
+
+    assert_json_include!(
+        actual: main_topic.next_event()?,
+        expected: json!({
+            "token": token,
+            "distinct_id": distinct_id1
+        })
+    );
+    assert_json_include!(
+        actual: main_topic.next_event()?,
+        expected: json!({
+            "token": token,
+            "distinct_id": distinct_id2
+        })
+    );
+
+    Ok(())
+}
+#[tokio::test]
+async fn it_captures_a_historical_batch() -> Result<()> {
+    setup_tracing();
+    let token = random_string("token", 16);
+    let distinct_id1 = random_string("id", 16);
+    let distinct_id2 = random_string("id", 16);
+
+    let main_topic = EphemeralTopic::new().await;
+    let histo_topic = EphemeralTopic::new().await;
+    let server = ServerHandle::for_topics(&main_topic, &histo_topic).await;
+
+    let event = json!({
+        "token": token,
+        "historical_migration": true,
```

Comment:
I was confused by the fact we have two flags: `is_historical` and `historical_migration` (for `/batch`).

Any reason we can't also use `is_historical` for `/batch`?

Reply:
The public API has been published already an is in use. I'll align the internal variables to `historical_migration` instead for consistency.

Reply:
Ah, okay, no problem then, happy to align internally. Thank you!